src/core/Class.js

```diff
@@ -13,21 +13,7 @@ export class Class {
 	// [Extends the current class](#class-inheritance) given the properties to be included.
 	// Returns a Javascript function that is a class constructor (to be called with `new`).
 	static extend({statics, includes, ...props}) {
-		const NewClass = function (...args) {
-
-			Util.setOptions(this);
-
-			// call the constructor
-			if (this.initialize) {
-				this.initialize.apply(this, args);
-			}
-
-			// call all constructor hooks
-			this.callInitHooks();
-		};
-
-		// inherit parent's prototype
-		Object.setPrototypeOf(NewClass.prototype, this.prototype);
+		const NewClass = class extends this {};

```

Comment:
We are now also actually using the `extends` keyword to extend parent classes properly.